File: app/src/main/java/com/poseidonapp/workorder/util/SharedPref.java
package com.poseidonapp.workorder.util;

import android.content.Context;
import android.content.SharedPreferences;

public class SharedPref {

    public static final String startDay="start_day";

    public static final String languageSelected="languageSelected";


    public static final String firesprinkler = "firesprinkler";

    private final SharedPreferences pref;

    private static SharedPref mInstance;

    public SharedPref(Context context) {
        mInstance = this;
        pref = context.getSharedPreferences("WorkOrder", 0);
    }


    public void setboolen(String name, boolean value){
        pref.edit().putBoolean(name,value).commit();
    }

    public boolean getboolen(String name) {
        return pref.getBoolean(name, false);
    }


    public void clear() {
        pref.edit().clear().commit();
    }

    //score updated today
    public void setScoreUpdated(boolean sprinker){
        //scoreUpdated
        pref.edit().putBoolean(firesprinkler,sprinker).commit();
    }

    public boolean getScoreUpdated(){
        return pref.getBoolean(firesprinkler,true);
    }


    public void setString(String name, String value){
        pref.edit().putString(name,value).commit();
    }

    public String getString(String name){
        return pref.getString(name,"");
    }

    public String getLanguageSelected(){
        return pref.getString(languageSelected, "En");
    }
    public void setLanguageSelected(String setLang){
        pref.edit().putString(languageSelected, setLang).commit();
    }


}

File: app/src/main/java/com/poseidonapp/utils/SharedPref.kt
package com.poseidonapp.utils

import android.content.Context
import android.content.SharedPreferences

class SharedPref(val context: Context) {
    var pref: SharedPreferences
    var edit: SharedPreferences.Editor



    init{
        pref=context.getSharedPreferences(Const.SHARED_PREF,0)
        edit=pref.edit()

    }

    public fun setString(key : String,value : String){
        edit.putString(key,value)
        edit.commit()
    }

    public fun getString(key : String):String{
        return pref.getString(key,"").toString()
    }

     fun setToken(key : String,value : String){
        edit.putString(key,value)
        edit.commit()
    }

    public fun getToken(key : String):String{
        return pref.getString(key,"").toString()
    }


    public fun setInt(key : String,value : Int){
        edit.putInt(key,value)
        edit.commit()
    }

    public fun getInt(key : String):Int{
        return pref.getInt(key,0)
    }


    public fun setBoolean(key : String,value : Boolean){
        edit.putBoolean(key,value)
        edit.commit()
    }

    public fun getBoolean(key : String):Boolean{
        return pref.getBoolean(key,false)
    }


    public fun getBooleanEstimatedTime(key : String):Boolean{
        return pref.getBoolean(key,true)
    }

    fun clearPref(){
        edit.clear().commit()
    }

    fun getLanguageSelected(): String? {
        return pref.getString(com.poseidonapp.workorder.util.SharedPref.languageSelected, "DARK")
    }

    fun setLanguageSelected(setLang: String?) {
        pref.edit().putString(com.poseidonapp.workorder.util.SharedPref.languageSelected, setLang).commit()
    }


    companion object {
      var  token = ""
    }
}

